tests/codemods/sonar/test_sonar_sql_parameterization.py

```diff
@@ -58,3 +63,51 @@ def f():
             ]
         }
         self.run_and_assert(tmpdir, input_code, expected, results=json.dumps(issues))
+
+    @pytest.mark.xfail(reason="TODO: See issue #607")
+    def test_more_complicated_example(self, tmpdir):
+        input_code = """
+        from django.shortcuts import redirect
+        from django.http import HttpResponse
+
+        import sqlite3
+        import json
+
+
+        def do_useful_things(request):
+            if request.method == "POST":
+                user = request.POST.get("user")
+
+                sql = "SELECT user FROM users WHERE user = '%s'"
+                conn = sqlite3.connect("example")
+                result = conn.cursor().execute(sql % user)
+
+                json_response = json.dumps({"user": result.fetchone()[0]})
+                return HttpResponse(json_response.encode("utf-8"))
+            else:
+                return redirect("/")
+        """.lstrip()
+        expected = """
+        from django.shortcuts import redirect
+        from django.http import HttpResponse
+
+        import sqlite3
+        import json
+
+
+        def do_useful_things(request):
+            if request.method == "POST":
+                user = request.POST.get("user")
+
+                sql = "SELECT user FROM users WHERE user = ?"
+                conn = sqlite3.connect("example")
+                result = conn.cursor().execute(sql, (user, ))
+
+                json_response = json.dumps({"user": result.fetchone()[0]})
+                return HttpResponse(json_response.encode("utf-8"))
+            else:
+                return redirect("/")
+        """.lstrip()
+
+        issues = json.dumps(SAMPLE_FILE_PATH.read_text())
```

Comment:
if the issue is still reproduced by just having issues = {...} as all other unit tests that would be a bit cleaner and easier to read which issue it is